src/logger.jl

```diff
@@ -117,6 +118,17 @@ end
 Similar to [`with_logging`](@ref), but append events to `l`.
 """
 function with_logging!(f, l::LogInfo)
+    # taskgraph must be empty before starting, or we may log dependencies on
+    # tasks that are not in the logger
+    tg = get_active_taskgraph()
+    _wait = !isempty(tg)
+    msg = """logging requires an empty taskgraph to start. Waiting for pending
+    tasks to be completed...
+    """
+    _wait && (@warn msg; wait(tg))
+    _wait && (@warn "done waiting.")
```

Comment:
Just a question of style, but is there a reason to avoid a (IMO) clearer construct like:
```julia
if !isempty(tg)
    msg = "waiting..."
    @warn msg
    wait(tg)
    @warn "done."
end
```